### DISCUSSION TITLE
Resolved.. not running optimized.. Performance not matching up. Router is 200 ms slower on average. Not Sending APQ to Downstream Services

### DISCUSSION THREAD

**Discussion by golfingal72** [apollographql/router#1191] [Q&A] [answered]
I am working with the new router and comparing performance and it's slower than the old gateway, even though documentation says it's much faster????

I am running locally all the processes in the graph as well as the gateway vs. the router.  I de-activated my gzip compression that i have running in the old gateway to try to do an apples to apples comparison...  I ran the same query which is large which has a large payload over and over again so ensure all caches are warmed up...  on average the new router is about 200 ms slower...  
I have to verify that the persisted query is getting sent to the backend services...  will follow up

**Comment by golfingal72** [comment]
Are you sending the APQ to the downstream services?

**Reply by golfingal72** [reply]
it appears the router is NOT sending the APQ downstream...  so this appears to be the performance issue

**Reply by o0Ignition0o** [reply]
Hey @golfingal72,

APQ is not sent downstream, we do have an issue tracking this [here](https://github.com/apollographql/router/issues/567).

**Comment by golfingal72** [comment, marked as answer]
Does it matter if i am running it as a cargo run command?

One of my peers suggested, it is faster to run by making a release executable?

**Reply by bnjjj** [reply]
Yes indeed you need to make a `cargo build --release` to have an optimized binary, otherwise you will have bad performances

**Reply by golfingal72** [reply]
That indeed was the PROBLEM...  I am new to Rust and didn't understand that ... thanks a bunch

**Reply by o0Ignition0o** [reply]
That's great news!
What kind of numbers are you seeing now?

**Reply by golfingal72** [reply]
i see it about 20 ms faster on my localhost...  haven't done extensive testing yet
